client/cmd/internal/plan/directory.go

```diff
@@ -0,0 +1,49 @@
+package plan
+
+type DirectoryParser interface {
+	ValidDirectory(directory string) bool
+	ParseDirectory(directory string) string
+}
+
+func GetValidJobDirectory(directories []string) []string {
+	var (
+		jobDirectories []string
+		jobParser      = new(JobPlan)
+	)
+
+	for i := range directories {
+		if jobParser.ValidDirectory(directories[i]) {
+			jobDirectories = append(jobDirectories, jobParser.ParseDirectory(directories[i]))
+		}
+	}
+
+	return jobDirectories
+}
+
+func GetValidResourceDirectory(directories []string) []string {
+	var (
+		resourcesDirectories []string
+		resourceParser       = new(ResourcePlan)
+	)
+
+	for i := range directories {
+		if resourceParser.ValidDirectory(directories[i]) {
+			resourcesDirectories = append(resourcesDirectories, resourceParser.ParseDirectory(directories[i]))
+		}
+	}
+
+	return resourcesDirectories
+}
+
+func DistinctDirectory(directories []string) []string {
+	directoryDictionary := make(map[string]bool)
+	res := make([]string, 0)
+	for i := range directories {
+		if directoryDictionary[directories[i]] {
```

Comment:
Can we check existence of key rather than value.
eg `if _, ok := directoryDictionary[dir]; ok {`

Reply:
updated @sbchaos https://github.com/goto/optimus/pull/228/commits/5d58f748810ebf732a569c26c7eed09daa94b3c2